Add support for EDS0066

tuomaz opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->
## Breaking change
<!--
  If your PR contains a breaking change for existing users, it is important
  to tell them what breaks, how to make it work again and why we did this.
  This piece of text is published with the release notes, so it helps if you
  write it towards our users, not us.
  Note: Remove this section if this PR is NOT a breaking change.
-->


## Proposed change
<!--
  Describe the big picture of your changes here to communicate to the
  maintainers why we should accept this pull request. If it fixes a bug
  or resolves a feature request, be sure to link to that issue in the
  additional information section.
-->
This change add support for EDS0066 (https://www.embeddeddatasystems.com/OW-ENV-TP--Temperature-Barometric-Pressure-Sensor_p_171.html) a variant of the already supported EDS0068 (https://www.embeddeddatasystems.com/OW-ENV-THPL--Temperature-Humidity-Barometric-Pressure-Light-Sensor_p_175.html). EDS0066 only has temperature and barometric sensors.

## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [X] New feature (which adds functionality to an existing integration)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes #
- This PR is related to issue: 
- Link to documentation pull request: https://github.com/home-assistant/home-assistant.io/pull/17711

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [X] The code change is tested and works locally.
- [ ] Local tests pass. **Your PR cannot be merged unless tests pass**
- [X] There is no commented out code in this PR.
- [X] I have followed the [development checklist][dev-checklist]
- [ ] The code has been formatted using Black (`black --fast homeassistant tests`)
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [ ] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [ ] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [ ] Untested files have been added to `.coveragerc`.

The integration reached or maintains the following [Integration Quality Scale][quality-scale]:
<!--
  The Integration Quality Scale scores an integration on the code quality
  and user experience. Each level of the quality scale consists of a list
  of requirements. We highly recommend getting your integration scored!
-->

- [ ] No score or internal
- [ ] 🥈 Silver
- [ ] 🥇 Gold
- [ ] 🏆 Platinum

<!--
  This project is very active and we have a high turnover of pull requests.

  Unfortunately, the number of incoming pull requests is higher than what our
  reviewers can review and merge so there is a long backlog of pull requests
  waiting for review. You can help here!
  
  By reviewing another pull request, you will help raise the code quality of
  that pull request and the final review will be faster. This way the general
  pace of pull request reviews will go up and your wait time will go down.
  
  When picking a pull request to review, try to choose one that hasn't yet
  been reviewed.

  Thanks for helping out!
-->

To help with the load of incoming pull requests:

- [ ] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[manifest-docs]: https://developers.home-assistant.io/docs/en/creating_integration_manifest.html
[quality-scale]: https://developers.home-assistant.io/docs/en/next/integration_quality_scale_index.html
[docs-repository]: https://github.com/home-assistant/home-assistant.io

tuomaz committed:
Add support for EDS0066

epenet commented:
Can you please add some tests?
You should be able to simply add a sensor in `tests\components\onewire\const.py`?

tuomaz commented:
Sure! I'll look into that.

tuomaz committed:
Added a test

tuomaz commented:
I've updated the PR. Added a sensor in const.py but had some problems installing all dependencies to tox (and hence running the tests). I'll try in a Linux env. (first try was on MacOS).

epenet commented:
@tuomaz I created the documentation PR.
Can you please edit the description to add the link? https://github.com/home-assistant/home-assistant.io/pull/17711

epenet commented on tests/components/onewire/const.py:
```diff
@@ -775,6 +775,36 @@
+        },
+        SENSOR_DOMAIN: [
+            {
+                "entity_id": "sensor.222222222222",
```
```suggestion
                "entity_id": "sensor.7e_222222222222_temperature",
```

epenet replied:
Still missing `7e` on the temperature sensor

epenet commented on tests/components/onewire/const.py:
```diff
@@ -775,6 +775,36 @@
+                "class": DEVICE_CLASS_TEMPERATURE,
+            },
+            {
+                "entity_id": "sensor.222222222222",
```
```suggestion
                "entity_id": "sensor.7e_222222222222_pressure",
```

epenet reviewed (changes requested):
Please correct the `entity_id` in the tests

epenet commented:
The link for the documentation should be added to the "Additional information" section of the PR description.
```
    Link to documentation pull request: xxxxxxxxxx
```

tuomaz committed:
Corrected entity_ids

tuomaz committed:
Added missing part of sensor entity id

epenet commented:
LGTM.
For the HA team: please enable workflow

epenet commented:
Test failure comes from ffmpeg - it is unrelated to this PR

epenet commented:
@tuomaz it looks like there is now a merge conflict.
Can you rebase?

tuomaz commented:
I tried to rebase but failed (got all the other changes in into this PR). What to do?

epenet commented:
The joys of GIT :( How to fix will depend on your GIT interface.

This is what I do on TortoiseGIT:
- switch to local branch
- fetch from upstream
- interactive rebase of local branch on upstream, picking only "my" commits

I don't know if there is a way to do this directly in the GitHub UI, nor do I know the commands for command line.

epenet commented on homeassistant/components/onewire/sensor.py:
```diff
@@ -195,7 +195,19 @@
 # 7E sensors are special sensors by Embedded Data Systems
 
-EDS_SENSORS: dict[str, list[DeviceComponentDescription]] = {
+EDS_SENSORS = {
```
This line is probably what caused the need to rebase originally.
Can you please reset it with the type hint?

epenet replied:
Thanks

epenet replied:
Thanks

tuomaz committed:
Add type hint

epenet commented:
LGTM (again).
For the HA team: please enable workflow (again)

janiversen commented on tests/components/onewire/const.py:
```diff
@@ -775,6 +775,36 @@
+        },
+        SENSOR_DOMAIN: [
+            {
+                "entity_id": "sensor.7e_222222222222_temperature",
```
is it "7e" or "7E" it is confusing.

I would have asked you to make a constant for "7E.222222222222", since it is repeated multiple times possibility or errors. However I can see you follow the pattern in the file so that change is for later.
```suggestion
                "entity_id": "sensor.7E_222222222222_temperature",
```

janiversen commented on tests/components/onewire/const.py:
```diff
@@ -775,6 +775,36 @@
+                "class": DEVICE_CLASS_TEMPERATURE,
+            },
+            {
+                "entity_id": "sensor.7e_222222222222_pressure",
```
```suggestion
                "entity_id": "sensor.7E_222222222222_pressure",
```
same here.

janiversen reviewed (changes requested):
Please add tests (as noted).

tuomaz committed:
Update tests/components/onewire/const.py

Co-authored-by: jan iversen <jancasacondor@gmail.com>

tuomaz committed:
Update tests/components/onewire/const.py

Co-authored-by: jan iversen <jancasacondor@gmail.com>

epenet commented:
@janiversen you change request causes the tests to fail.
The casing is different for the device listed on onewire (can't be changed) and the sensor entity id (standardised as lower case). This is on purpose.

epenet commented:
Also the tests are implemented. Not sure why you are asking for extra tests.

janiversen commented:
> @janiversen your change request will (and does) cause the tests to fail.
> The casing is different for the device listed on onewire (comes from 3rd party, and can't be changed) and the sensor entity id (standardised as lower case). This is on purpose.

 I did not know that, and it was not clear looking at the code, maybe a one line comment can avoid others get confused.

janiversen commented:
> Also the tests are implemented. Not sure why you are asking for extra tests.

I added that line because I saw this cooment:
“ Can you please add some tests?
You should be able to simply add a sensor in tests\components\onewire\const.py?”
and did not see a confirmation, you also have a couple of unresolved conversations (which I think are resolved but not marked as such)

epenet commented:
@tuomaz can you please revert the changes from 151eb9c0d3303b6bd3b3dc49a1eccd7c1a1b31b8 and 4a01b89868bb692bb2911ca5b9f9939611a5ff2f?

@janiversen we can look at the constants in a separate PR. Regarding the conversation marked as resolved I can't do it (maybe I don't have rights to do it). @tuomaz can mark them as resolved...

tuomaz committed:
Revert "Update tests/components/onewire/const.py"

This reverts commit 4a01b89868bb692bb2911ca5b9f9939611a5ff2f.

tuomaz committed:
Revert "Update tests/components/onewire/const.py"

This reverts commit 151eb9c0d3303b6bd3b3dc49a1eccd7c1a1b31b8.

tuomaz commented:
I think I managed to revert the latest changes.

epenet commented:
Thanks @tuomaz.
Can you please mark the previous comments as "resolved" so other are not confused?

epenet commented:
HA Team: can we re-enable the workflow and get this merged in for 2021.6 ?

tuomaz commented:
Have now marked as resolved.

janiversen reviewed (approved):
Looks good, thanks.

epenet commented:
Thanks @janiversen
Do you want to take a look at similar PR #50148?
